pennylane/qchem/observable_hf.py

```diff
@@ -135,13 +136,17 @@ def qubit_observable(o_ferm, cutoff=1.0e-12, mapping="jordan_wigner"):
     """
     if mapping == "jordan_wigner":
         h = qml.jordan_wigner(o_ferm, ps=True, tol=cutoff)
+
     elif mapping == "parity":
         qubits = len(o_ferm.wires)
         h = qml.parity_transform(o_ferm, qubits, ps=True, tol=cutoff)
     elif mapping == "bravyi_kitaev":
         qubits = len(o_ferm.wires)
         h = qml.bravyi_kitaev(o_ferm, qubits, ps=True, tol=cutoff)
 
+    if list(h.wires) != sorted(list(h.wires)):
+        h = sum(i * h[i] for i in sorted(h.keys(), key=lambda x: list(x.wires)))
```

Comment:
![image](https://github.com/PennyLaneAI/pennylane/assets/23234279/10816a44-948b-4439-a3aa-d15e3be71dd5)

Efficiency-wise, it is faster to do the above here (tested for H4):
```
qml.pauli.PauliSentence(sorted(h.items(), key=lambda item: item[0].wires.tolist()))
```